core/CHANGELOG.md

```diff
@@ -36,6 +36,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+* [#17693](https://github.com/cosmos/cosmos-sdk/pull/17693) Remove `appmodule.UpgradeModule` interface in favor of preblock
```

Comment:
As core v0.12 at the time or writing hasn't yet been tagged, this could just be included in that release as well.